src/authenticators.rs

```diff
@@ -48,6 +48,36 @@ impl Authenticator for NoneAuthenticator {
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum GeneralAuthenticator {
```

Comment:
I don't think it's needed, because we already accept `dyn Authenticator`.

Reply:
That's fair. I can drop this patch from the PR.